### PR TITLE
NMS-10067: Enhance AlarmPersisterImpl to support updating acknowledgment values on reduction

### PR DESCRIPTION
**Pull request by jeffgdotorg**
* JIRA: http://issues.opennms.org/browse/NMS-10067

### CODE REVIEW

File: opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java [hunk 1]
```diff
@@ -63,6 +63,7 @@
  */
 public class AlarmPersisterImpl implements AlarmPersister {
     private static final Logger LOG = LoggerFactory.getLogger(AlarmPersisterImpl.class);
+    public static final String EVENT_SOURCE_NAME = "Alarmd";
```
Comment (j-white) [inline opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java]: Unused constant.

File: opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java [hunk 2]
```diff
@@ -174,7 +175,7 @@ private OnmsAlarmAndLifecycleEvent addOrReduceEventAsAlarm(Event event, boolean
         return new OnmsAlarmAndLifecycleEvent(alarm, ebldr.getEvent());
     }
 
-    private static void reduceEvent(OnmsEvent e, OnmsAlarm alarm, Event event) {
+    private static void reduceEvent(OnmsEvent e, OnmsAlarm alarm, Event event, EventForwarder f) {
```
Comment (j-white) [inline opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java]: Unused argument.

File: opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java [hunk 3]
```diff
@@ -211,7 +212,8 @@ private static void reduceEvent(OnmsEvent e, OnmsAlarm alarm, Event event) {
                         alarm.setSeverity(OnmsSeverity.valueOf(e.getSeverityLabel()));
                     } else if (fieldName.toLowerCase().contains("descr")) {
                         alarm.setDescription(e.getEventDescr());
-                        alarm.setSeverity(OnmsSeverity.valueOf(e.getSeverityLabel()));
```
Comment (j-white) [inline opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java]: Was this removal intentional?

**Comment by pbrane** [comment]
Those "unused" issues references should be deleted.  This was from the patch I provided.
The update of description when the update of severity was being handled appeared to be a bug IMHO since that removes control of that field from the user's configuration.  Probably something I did decades ago. :(

File: opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java (hunk 3 above)
Comment (jeffgdotorg) [reply opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java]: I agree with @pbrane that this one is likely a copypasta bug.

**Comment by pbrane** [comment]
Let's do that later as another enhancement issue.

**Comment by j-white** [comment]
@jeffgdotorg I'd opt for the JEXL expression which would give more flexibility.

Users using this feature may also want to update the `unclear` automation to un-ack as well.

Once merged, can we also look at updating the wiki page so that this is reference-able somewhere other than in the code: https://wiki.opennms.org/wiki/Configuring_alarms#Update_Field
